spec/services/token_manager_spec.rb

```diff
@@ -0,0 +1,144 @@
+require 'rails_helper'
+
+RSpec.describe TokenManager do
+  subject{ TokenManager }
+  let(:user){ create(:user) }
+
+  describe ".generate_token" do
+    let(:token){ subject.generate_token(user.id) }
+
+    it 'generates a valid token' do
+      expect(token.split('.').count).to eq 3
+      expect{subject.decode(token)}.not_to raise_error
+    end
+
+    describe 'token expiration' do
+      it 'sets expiration time to 24hrs by default' do
+        decoded = subject.decode(token)
+        expect(decoded.first['exp']).to be_within(100).of(24.hours.from_now.to_i)
```

Comment:
Thanks for pointing out the `delorean` gem. It's awesome, especially given that writing tests that involve time are tricky.
Using it here, however, might seem like overflogging the issue. The need for the `be_within` matcher is to allow for any form system drag or delay while running the test which might affect the calculated expectation of the test.
I'm all for reducing the time to say `2` seconds; but I once experienced up to *72* seconds while experiment locally.